dask/array/slicing.py

```diff
@@ -1240,3 +1240,29 @@ def cached_cumsum(seq, initial_zero=False):
     if not initial_zero:
         result = result[1:]
     return result
+
+
+def take_along_axis(arr,indices,axis):
+    from .core import from_array
+    out_d = np.array([]) #creating a new empty array to store the value of result
+    arr = np.array(arr)
+    indices = np.array(indices)
+    if arr.shape != indices.shape:
+        raise ValueError(
+            "`indices` and `arr` must have the same number of dimensions")
+    if (indices.dtype != 'int64'):
+        raise IndexError('indices must be an integer array')
+    if axis == 0:
+        arr = arr.flatten
```

Comment:
Do you need to call this?
```suggestion
        arr = arr.flatten()
```

Reply:
I think `arr.flatten` will work fine but I am unable to rectify the problem regarding this one

> On running pytest I am getting :
> `AttributeError: 'builtin_function_or_method' object has no attribute 'shape'` for line `Ni, Nk = arr.shape[:axis], arr.shape[axis + 1:]`

Reply:
`flatten` is a method. You need to call it, right?

```python
In [54]: arr = np.array([1,2 , 3])

In [55]: arr.flatten
Out[55]: <function ndarray.flatten>

In [56]: arr.flatten()
Out[56]: array([1, 2, 3])
```

Reply:
Thank you for correcting me can you also help me on this

> I think `arr.flatten` will work fine(similar thing has been done in numpys `take_along_axis`) but I am unable to rectify the problem regarding this one
> 
> > On running pytest I am getting :
> > `AttributeError: 'builtin_function_or_method' object has no attribute 'shape'` for line `Ni, Nk = arr.shape[:axis], arr.shape[axis + 1:]`

Reply:
Won't be able to take a look for a while.